modules/ipc/ipcProviderBackend.js

```diff
@@ -288,66 +294,43 @@ class IpcProviderBackend {
             } else {
                 event.sender.send('ipcProvider-data', returnValue);
             }
-        })
+        });        
     }
 
 
+
     /**
-    Validate a request payload.
+    Sanitize a single or batch request payload.
+
+    This will modify the passed-in payload.
 
-    @method _validateRequestPayload
     @param {Object} conn The connection.
-    @param {Object} payload The request payload.
-    @throw {Error} if request invalid.
+    @param {Object|Array} payload The request payload.
     */
-    _validateRequestPayload(conn, payload) {
-        log.trace('Filter request payload', payload);
-
-        // main window or popupwindows - always allow requests
-        let wnd = Windows.getById(conn.id);
-
-        if (wnd && ('main' === wnd.type || wnd.isPopup)) {
-            return payload;
-        }
-
-        let __check = (p) => {
-            if (!_.isObject(p)) {
-                throw ERRORS.INVALID_PAYLOAD;
-            }
-
-            // prevent dapps from acccesing admin endpoints
-            if(!/^eth_|^shh_|^net_|^web3_|^db_/.test(p.method)){
-                throw ERRORS.METHOD_DENIED;
-            }
-        }
-
-
+    _sanitizeRequestPayload (conn, payload) {
         if (_.isArray(payload)) {
-            for (let p of payload) {
+            _.each(payload, (p) => {
                 if ('eth_sendTransaction' === p.method) {
-                    throw ERRORS.BATCH_TX_DENIED;                    
+                    p.error = ERRORS.BATCH_TX_DENIED;
```

Comment:
Adding the error object alone, is not enough, you would need to remove the `method` and `params` property too.

Maybe adding a clean result functions somewhere later?